### PR TITLE
feat(perf): add AngularDart v1 of table scrolling benchmark

### PR DESCRIPTION
**Pull request by yjbanov**
The benchmark is a naive implementation of an infinite scroller. It destroys and creates rows as your scroll. Use query parameter appSize to indicate how many tables to draw on the UI. This is useful to measure scalability of the digest loop.

### CODE REVIEW

File: modules/benchmarks_external/src/naive_infinite_scroll/app.dart [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    ''')
+class App implements ShadowRootAware {
+  final VmTurnZone ngZone;
+  List<int> scrollAreas;
+  int scrollTop = 0;
+  String result;
+
+  App(this.ngZone) {
+    int appSize = 1;
+    var location = Uri.parse(window.location.href);
```
Comment (tbosch) [inline modules/benchmarks_external/src/naive_infinite_scroll/app.dart]: Could you use the methods from `e2e_test_lib/benchmark_util` for reading parameters and for binding to buttons as the other benchmarks do?

And we have the default value for the parameter in the html...

File: modules/benchmarks_external/src/naive_infinite_scroll/app.dart [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    if (location.queryParameters.containsKey('appSize')) {
+      appSize = int.parse(location.queryParameters['appSize']);
+    }
+    appSize = appSize > 1 ? appSize - 1 : 0;
+    scrollAreas = new List.generate(appSize, (i) => i);
+  }
+
+  @override
+  void onShadowRoot(ShadowRoot shadowRoot) {
+    ButtonElement runBtn = shadowRoot.querySelector('#run-btn');
```
Comment (tbosch) [inline modules/benchmarks_external/src/naive_infinite_scroll/app.dart]: Use `benchmark_util.bindAction`, see above

File: modules/benchmarks_external/src/naive_infinite_scroll/app.dart [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    runBtn.onClick.listen((_) {
+      runBenchmark();
+    });
+  }
+
+  void runBenchmark() {
+    result = 'Running benchmark...';
+    var sw = new Stopwatch()..start();
```
Comment (tbosch) [inline modules/benchmarks_external/src/naive_infinite_scroll/app.dart]: If you create a benchpress test we don't need the StopWatch any more. For manually executing the test, people should look at the Timeline, for automatically reading the value we use the performance logs.

**Comment by tbosch** [comment]
Could you also create a simple perf test with benchpress?

File: modules/benchmarks_external/src/naive_infinite_scroll/app.dart (hunk 1 above)
Comment (yjbanov) [inline modules/benchmarks_external/src/naive_infinite_scroll/app.dart]: done

File: modules/benchmarks_external/src/naive_infinite_scroll/app.dart (hunk 3 above)
Comment (yjbanov) [inline modules/benchmarks_external/src/naive_infinite_scroll/app.dart]: done

**Comment by yjbanov** [comment]
Done everything, I think.

File: modules/benchmarks_external/src/naive_infinite_scroll/app.dart (hunk 2 above)
Comment (yjbanov) [inline modules/benchmarks_external/src/naive_infinite_scroll/app.dart]: done

File: modules/benchmarks_external/e2e_test/naive_infinite_scroll_perf.es6 [hunk 4]
```diff
@@ -0,0 +1,33 @@
+var perfUtil = require('../../e2e_test_lib/e2e_test/perf_util');
+
+describe('ng-dart1.x naive infinite scroll benchmark', function () {
+
+  var URL = 'benchmarks_external/web/naive_infinite_scroll/index.html';
+
+  afterEach(perfUtil.verifyNoBrowserErrors);
+
+  [2, 4, 8, 16].forEach(function(appSize) {
+    iit('should run scroll benchmark and collect stats for appSize = ' +
```
Comment (tbosch) [inline modules/benchmarks_external/e2e_test/naive_infinite_scroll_perf.es6]: :-)

Comment (yjbanov) [inline modules/benchmarks_external/e2e_test/naive_infinite_scroll_perf.es6]: done

File: modules/benchmarks_external/e2e_test/naive_infinite_scroll_spec.es6 [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+  afterEach(testUtil.verifyNoBrowserErrors);
+
+  it('should not throw errors', function() {
+    browser.get(URL);
+    browser.executeScript(
+        'document.querySelector("scroll-app /deep/ #reset-btn").click()');
+    browser.executeScript(
+        'document.querySelector("scroll-app /deep/ #run-btn").click()');
+  });
```
Comment (tbosch) [inline modules/benchmarks_external/e2e_test/naive_infinite_scroll_spec.es6]: I would add a `sleep(1000)` here as well to ensure that the scoll completed (just as above)

File: modules/e2e_test_lib/e2e_test/perf_util.es6 [hunk 6]
```diff
@@ -3,10 +3,22 @@ var webdriver = require('protractor/node_modules/selenium-webdriver');
...
 
 function runClickBenchmark(config) {
+  var buttons = config.buttons.map(function(selector) {
+    return $(selector);
+  });
+  config.work = function() {
+    buttons.forEach(function(button) {
+      button.click();
+    });
+  }
```
Comment (tbosch) [inline modules/e2e_test_lib/e2e_test/perf_util.es6]: I think you want to call `runBenchmark(config)` here...

**Comment by tbosch** [comment]
LGTM, please fix up and merge.

File: modules/e2e_test_lib/e2e_test/perf_util.es6 (hunk 6 above)
Comment (yjbanov) [inline modules/e2e_test_lib/e2e_test/perf_util.es6]: done

File: modules/benchmarks_external/e2e_test/naive_infinite_scroll_spec.es6 (hunk 5 above)
Comment (yjbanov) [inline modules/benchmarks_external/e2e_test/naive_infinite_scroll_spec.es6]: done

**Pull request merged**